### ISSUE TITLE
Request Joined MaxLE/MaxASP/MaxAu with Current LE/ASP/Au

### ISSUE THREAD

**Issue by Necrask** [Roll20/roll20-character-sheets#8720]
Hello,

I'm playing around with tokens at the moment, and saw that you can choose charaktervalues for the token. With the current system you can either choose to add only the Max Value to the token 0/Value or just the current value, which will automatically change the current value on the token and the char sheet but since this value doesn't have an max value, it wont automatically generate a bar. (See picture).
![Example](https://user-images.githubusercontent.com/67807029/114681540-8ae07180-9d0e-11eb-8a83-6daba72709a9.png)

So I would suggest another Attribute named Current/MaxLE. Current/MaxASP and Current/MaxAu where the two values are joined in one attribute.

If my request isn't clear, I can rewrite it in german.

Thanks!

**Comment by Stexinator** [comment]
@Necrask please put the sheet name in the description and title. Noone can guess what you are talking about from just some attributes

**Comment by Necrask** [comment]
Sorry, my bad.

It isn't exactly how Roll20 uses tokens, it about using tokens with the DSA4.1 charakter sheet without manual typing either the current values (without overwrite into the charakter sheet) or maximum values into the token.
The attributes on Roll20 have two values, for current the 1st of the two is used and for max the 2nd, but in two different attributes. If there were an attribute where both values were combined, one click would give the current/max in the token.

**Comment by Necrask** [comment]
This is what I meant in the post before and which values are taken from the token generator.
![Testsheet](https://user-images.githubusercontent.com/67807029/114735230-445a3980-9d45-11eb-86f8-e924ce04510c.png)
![Testsheet1](https://user-images.githubusercontent.com/67807029/114735234-44f2d000-9d45-11eb-90c7-b52cf84228f9.png)

**Comment by Stexinator** [comment]
@kreuvf for visibility

**Comment by kreuvf** [comment]
> If my request isn't clear, I can rewrite it in german.

Please do so, I think I understand, but I want to make sure I did not get you wrong. :)

**Comment by Necrask** [comment]
Alles klar,

also die Token auf Roll20 können ja bis zu drei Bars haben. Diesen Bars kann man wie im oberen Bild gezeigt Charaktersheetwerte(Attribute) zu ordnen. Damit die Bars allerdings vollständig angezeigt werden braucht man den Aktuellen Wert und den Maximalwert in einem gemeinsamen Attribut.

Der weitere Vorteil der zuweisung von Attributen zum Token ist, dass wenn man z.B. sich beim Token Lebenspunkte abzieht, wird dies automatisch auf den Charakterbogen übertragen, so dass man dies nicht immer wieder hin und her übertragen muss.

Aktuell gibt es allerdings Aktuelle Werte und Maxwerte nur als getrennte Attribute. Deshalb meine Nachfrage, ob man nicht für LE, Aus und ASP je ein weiteres Attribut hinzufügen kann im Code, das Current und Max kombiniert - Also mittlere Zelle Current, rechte Zelle Max. (Bild 3).

Ich hoffe dies war verständlicher?
Ansonsten kann ich noch ein, zwei Screens nachfügen.

**Comment by kreuvf** [comment]
> Alles klar,
> 
> also die Token auf Roll20 können ja bis zu drei Bars haben. Diesen Bars kann man wie im oberen Bild gezeigt Charaktersheetwerte(Attribute) zu ordnen. Damit die Bars allerdings vollständig angezeigt werden braucht man den Aktuellen Wert und den Maximalwert in einem gemeinsamen Attribut.

Das ist richtig.

> Der weitere Vorteil der zuweisung von Attributen zum Token ist, dass wenn man z.B. sich beim Token Lebenspunkte abzieht, wird dies automatisch auf den Charakterbogen übertragen, so dass man dies nicht immer wieder hin und her übertragen muss.

Ja, kenne ich.

> Aktuell gibt es allerdings Aktuelle Werte und Maxwerte nur als getrennte Attribute. Deshalb meine Nachfrage, ob man nicht für LE, Aus und ASP je ein weiteres Attribut hinzufügen kann im Code, das Current und Max kombiniert - Also mittlere Zelle Current, rechte Zelle Max. (Bild 3).

Ich verstehe das Problem. Wir nutzen in unserer Gruppe das Sheet und können über die Tokens LE, AU, AE verändern. Den Wert für das Maximum haben wir händisch eingetragen (ja, mal wieder einer dieser Workarounds ^_^). Als Attribute bitte die folgenden verwenden:
![image](https://user-images.githubusercontent.com/518136/115048509-edd23400-9ed9-11eb-8767-ae40f133a911.png)

LE: LEAktuell
AU: AusAktuell
AE: ASPAktuell

Du siehst an der Benennung der Attribute, dass die Autoren die automatisch von Roll20 eingebauten Attribute mit der Endung "_max" nicht berücksichtigt haben und stattdessen eigene Attribute mit "_max" eingebaut haben. Das passt natürlich nicht zusammen und sorgt für dein Problem. Ich nehme das mal in ein Issue auf, weil das tatsächlich ein nerviges Problem ist.

Siehe neues Issue: https://github.com/kreuvf/roll20-character-sheets/issues/70.

 @nmbradley: This will be resolved in a future PR. I opened an issue on my fork for this.

**Comment by Necrask** [comment]
Den Workaround hatte ich schon gefunden und er funktioniert auch. Meine Frage war, ob man nicht ein zusätzliches Attribut einfügen kann, dass Aktuell und Max in eines kombiniert, sodass man mit den Token einfach darauf zugreifen kann und nicht mehr den Workaround nutzen muss, die max Werte per Hand hinzuzufügen.

**Comment by kreuvf** [comment]
> Den Workaround hatte ich schon gefunden und er funktioniert auch. Meine Frage war, ob man nicht ein zusätzliches Attribut einfügen kann, dass Aktuell und Max in eines kombiniert, sodass man mit den Token einfach darauf zugreifen kann und nicht mehr den Workaround nutzen muss, die max Werte per Hand hinzuzufügen.

Ja, ich will das einbauen und es ist auch das Ziel, diese Funktionalität richtig/vollständig zu unterstützen :) Daher auch das eigene Issue nur dafür ^_^

**Comment by kreuvf** [comment]
> Den Workaround hatte ich schon gefunden und er funktioniert auch. Meine Frage war, ob man nicht ein zusätzliches Attribut einfügen kann, dass Aktuell und Max in eines kombiniert, sodass man mit den Token einfach darauf zugreifen kann und nicht mehr den Workaround nutzen muss, die max Werte per Hand hinzuzufügen.

Das hatte ich ganz überlesen xD Ich wüsste aus dem Kopf nicht, wie man das anstellen könnte. Ich rate auch grundsätzlich davon ab, händisch an den Attributen herumzuspielen, wenn es nicht notwendig ist. Die Folgen können schwer zu entdeckende Fehler sein ^.^

**Comment by nmbradley** [comment]
@Necrask @kreuvf Is this resolved?

**Comment by kreuvf** [comment]
> @Necrask @kreuvf Is this resolved?

Not yet. It is on my list of issues to tackle next.
